order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java

```diff
@@ -70,10 +86,161 @@
 			return new ResponseEntity<>(new ApiErrorsWS(errorWSList), HttpStatus.UNPROCESSABLE_ENTITY);
 		}
 
-//		OrderBean orderBean = orderMapper.toBean(orderWS);
-//		OrderBean order = orderService.createOrder(orderBean);
+		OrderBean orderBean = orderMapper.toBean(orderWS, new CycleAvoidingMappingContext());
+
+		OrderBean order = orderService.createOrder(orderBean);
+
+		return new ResponseEntity<>(orderMapper.toWS(order, new CycleAvoidingMappingContext()), HttpStatus.OK);
+	}
+
+
+	@PutMapping("/order/{uuid}")
+	@RequiresJwtAuthentication
+	@RequiresPermissions("order:update")
+	public ResponseEntity<?> updateOrder(@PathVariable("uuid") String orderUuid, @Validated @RequestBody OrderWS orderWS) {
+
+		Optional<OrderBean> tmpOrderBean = orderService.findByUuid(orderUuid);
+
+		if(!tmpOrderBean.isPresent()) {
+			log.error("Error in update order. Fail in finding order uuid={}", orderUuid);
+			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.ORDER_UUID_NOT_FOUND.name(), "Cannot find order by uuid"), HttpStatus.NOT_FOUND);
+		}
+
+		ResponseEntity<?> validateUserAgentUuidResult = validateUserAndAgentUuid(orderWS);
+
+		if(validateUserAgentUuidResult != null) {
+			return validateUserAgentUuidResult;
+		}
+
+		BindException errors = new BindException(orderWS, "OrderWS");
+
+		orderValidator.validate(orderWS, errors);
+
+		if (errors.hasErrors()) {
+			log.error("Error in update order. Fail in update order validation fields. orderWS={}", orderWS.toString());
+			List<ErrorWS> errorWSList = apiErrorsExceptionHandler.generateErrorWSList(errors);
+			return new ResponseEntity<>(new ApiErrorsWS(errorWSList), HttpStatus.UNPROCESSABLE_ENTITY);
+		}
+
+		OrderBean order = tmpOrderBean.get();
+
+		OrderBean orderBean = orderMapper.toBean(orderWS, new CycleAvoidingMappingContext());
+
+		orderBean.setId(order.getId());
+
+		order = orderService.updateOrder(orderBean);
+
+		return new ResponseEntity<>(orderMapper.toWS(order, new CycleAvoidingMappingContext()), HttpStatus.OK);
+	}
+
+	@DeleteMapping("/order/{uuid}")
+	@RequiresJwtAuthentication
+	@RequiresPermissions("order:update")
+	public ResponseEntity<?> deleteOrder(@PathVariable("uuid") String orderUuid) {
 
+		Optional<OrderBean> order = orderService.findByUuid(orderUuid);
+
+		if(!order.isPresent()){
+			log.error("Error in delete order. Fail in finding order uuid={}", orderUuid);
+			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.ORDER_UUID_NOT_FOUND.name(), "Cannot find order by uuid that is not found"), HttpStatus.NOT_FOUND);
+		}
 
+		orderService.deleteOrder(order.get());
+
+		return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+	}
+
+	@GetMapping("/order/{uuid}")
+	@RequiresJwtAuthentication
+	@RequiresPermissions("order:read")
+	public ResponseEntity<?> readOrder(@PathVariable("uuid") String orderUuid) {
+
+		Optional<OrderBean> order = orderService.findByUuid(orderUuid);
```

Comment:
`OrderRepository` and have `findByUuidAndDeleted(String, boolean)`. The service layer always getOrder with deleted == false.

Reply:
fixed